packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -43,48 +69,63 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
        * TODO: Fix the types for the middleware
        */
       strapi.documents.use(async (context, next) => {
-        // @ts-expect-error ContentType is not typed correctly on the context
-        const contentTypeUid = context.contentType.uid;
-        const params = context.args.at(-1) as any;
+        // Ignore requests that are not related to the content manager
+        if (!strapi.requestContext.get()?.request.url.startsWith('/content-manager')) {
+          return next(context);
+        }
+
         // Ignore actions that don't mutate documents
-        if (!['create', 'update', 'publish', 'unpublish'].includes(context.action)) {
+        if (
+          !['create', 'update', 'publish', 'unpublish', 'discardDraft'].includes(context.action)
+        ) {
           return next(context);
         }
 
+        // @ts-expect-error ContentType is not typed correctly on the context
+        const contentTypeUid = context.contentType.uid;
         // Ignore content types not created by the user
         if (!contentTypeUid.startsWith('api::')) {
           return next(context);
         }
 
+        const result = (await next(context)) as any;
+
+        const documentContext =
+          context.action === 'create'
+            ? // @ts-expect-error The context args are not typed correctly
+              { documentId: result.documentId, locale: context.args[0]?.locale }
+            : { documentId: context.args[0], locale: context.args[1]?.locale };
+
+        const locale = documentContext.locale ?? localesService.getDefaultLocale();
+        const document = await strapi
+          .documents(contentTypeUid)
+          .findOne(documentContext.documentId, {
+            locale,
+            populate: '*',
```

Comment:
Is there a way to populate only the ID everywhere? Because we'll never use the rest of the data, and it can take a lot of space for nothing

Reply:
oh yeah I'm sure I can just pick what we need here

Reply:
I take that back, I'm not sure I can populate all relation but only return the ID 🤔

Reply:
huh, that's too bad. And without populate I guess there's just no object entry at all?

Reply:
you could do some logic to trim the data before saving, but it's a bit annoying

Reply:
or you write a script that creates the populate object based on the schema (to identify where there are relations/images). Just as annoying, but faster

Reply:
I can actually remove the populate here. It is unrelated to handling the status 🙃 .

Reply:
but document ends up in the data column of the history version though right?

Reply:
if it is still needed, @Marc-Roig sent me this file which seems very handy: https://github.com/strapi/strapi/blob/f7d9269c38f7d4b8590c6f7b5c15981814c6d662/packages/core/core/src/services/document-service/utils/populate.ts

Reply:
- A populate object is sent along with the context so maybe we can use that. 
![Screenshot 2024-03-14 at 16 13 58](https://github.com/strapi/strapi/assets/26598053/037951ec-850b-46f5-8ac1-6ee18d577b5a)
- Even with populate removed here the statuses display correctly and the document ends up in the version's data column, it just doesn't include relations. Which I believe is more or less how it is now on v5/main.

Reply:
>the document ends up in the version's data column, it just doesn't include relations

yes but I think we need them though 🤔 It's like that currently on main because we never addressed relations, but I'm currently working on that, and I just need to know the IDs of related entries

Reply:
regarding the populate object, are we sure it will always be there? from what we've seen the context data is not very consistent, I'd prefer not to rely on it too much

and we don't need all the default fields (strapi_stage, strapi_assignee, created/updatedBy), only the ones that were created by the user

Reply:
> yes but I think we need them though 🤔 It's like that currently on main because we never addressed relations, but I'm currently working on that, and I just need to know the IDs of related entries

Right but should that be handled by this PR which is about "status"? In my opinion it feels out of scope and should be in a PR that is about relations. Let me know if I'm not understanding something because I'm confused why it should be done here. 

> regarding the populate object, are we sure it will always be there? from what we've seen the context data is not very consistent, I'd prefer not to rely on it too much

Yes I agree

Reply:
You're right I can address it in my PR, it's very related to what I'm doing now